mlrun/utils/helpers.py

```diff
@@ -2423,6 +2429,45 @@ def set_data_by_path(
         )
 
 
+def _normalize_requirements(reqs: typing.Union[str, list[str], None]) -> list[str]:
+    if reqs is None:
+        return []
+    if isinstance(reqs, str):
+        s = reqs.strip()
+        return [s] if s else []
+    return [s.strip() for s in reqs if s and s.strip()]
+
+
+def merge_requirements(
+    reqs_priority: typing.Union[str, list[str], None],
+    reqs_secondary: typing.Union[str, list[str], None],
+) -> list[str]:
+    """
+    Merge two requirement collections into a union. If the same package
+    appears in both, the specifier from reqs_secondary wins.
+
+    Args:
+        reqs_priority: str | list[str] | None  (priority input)
+        reqs_secondary: str | list[str] | None
+
+    Returns:
+        list[str]: pip-style requirements.
+    """
+    merged: dict[str, Requirement] = {}
+
+    for r in _normalize_requirements(reqs_priority):
```

Comment:
```suggestion
    for r in _normalize_requirements(reqs_secondary) + _normalize_requirements(reqs_primary):
```
If you do this, you can just keep this loop, and get rid of the other one. Since the primary will get processed after the secondary, then overrides will happen automatically once you set it to the same name in the `merged` dict.